packages/react-client/src/ReactFlightClient.js

```diff
@@ -189,6 +208,18 @@ function resolveModelChunk<T>(
   wakeChunk(listeners);
 }
 
+function resolveProviderChunk<T>(chunk: SomeChunk<T>, value: string): void {
+  if (chunk._status !== PENDING) {
+    // We already resolved. We didn't expect to see this.
+    return;
+  }
+  const listeners = chunk._value;
+  const resolvedChunk: ResolvedProviderChunk = (chunk: any);
+  resolvedChunk._status = RESOLVED_PROVIDER;
+  resolvedChunk._value = value;
+  wakeChunk(listeners);
```

Comment:
Look at what we do for Symbols instead. They go straight from resolve to initialized.

https://github.com/facebook/react/blob/main/packages/react-client/src/ReactFlightClient.js#L391

That saves us some indirection and code.

The reason the other ones don't is because they might be large and take some CPU time to parse etc. Therefore we do that lazily only when we actually start needing those objects. That way we're better able to prioritize the CPU work and if it doesn't end up getting rendered we don't need it.

For Symbols and Providers, the parsing/reading from a map is trivial. So we might as well bypass the laziness.